applications/asset_tracker_v2/src/modules/location_module.c

```diff
@@ -0,0 +1,631 @@
+	 * and the time that the application will wait for A-GPS data before including GNSS
+	 * in sample requests can be adjusted via the
+	 * CONFIG_APP_REQUEST_GNSS_WAIT_FOR_AGPS_THRESHOLD_SEC Kconfig option. If this option is
+	 * set to -1, the application module will not request GNSS unless A-GPS data has been
+	 * processed.
+	 *
+	 * If A-GPS data has been processed the sample timeout can be ignored as the
+	 * GNSS will most likely time out before the sample timeout expires.
+	 *
+	 * When GNSS is requested, set the sample timeout to (GNSS timeout + 15 seconds)
+	 * to let the GNSS module finish ongoing searches before data is sent to cloud.
+	 */
+	if (methods_index_gnss != -1) {
+		config.methods[methods_index_gnss].gnss.timeout =
+			MAX(copy_cfg.gnss_timeout + 15, 75) * MSEC_PER_SEC;
+	}
+	if (methods_index_cellular != -1) {
+		config.methods[methods_index_cellular].cellular.timeout =
+			DATA_FETCH_TIMEOUT_NEIGHBORHOOD_SEARCH * MSEC_PER_SEC;
+	}
+
+	LOG_DBG("Requesting location...");
+
+	err = location_request(&config);
+	if (err) {
+		// TODO:
+		// SEND_EVENT(modem, LOCATION_EVT_LOCATION_DATA_NOT_READY);
+		LOG_ERR("Location request failed: %d", err);
+		return;
+	}
+	SEND_EVENT(location, LOCATION_MODULE_EVT_ACTIVE);
+	stats.start_uptime = k_uptime_get();
+}
+
+static void inactive_send(void)
+{
+	SEND_EVENT(location, LOCATION_MODULE_EVT_INACTIVE);
+}
+
+static void time_set(void)
+{
+	/* Change datetime.year and datetime.month to accommodate the
+	 * correct input format.
+	 */
+	struct tm gnss_time = {
+		.tm_year = pvt_data.datetime.year - 1900,
+		.tm_mon = pvt_data.datetime.month - 1,
+		.tm_mday = pvt_data.datetime.day,
+		.tm_hour = pvt_data.datetime.hour,
+		.tm_min = pvt_data.datetime.minute,
+		.tm_sec = pvt_data.datetime.seconds,
+	};
+
+	date_time_set(&gnss_time);
+}
+
+static void location_event_handler(const struct location_event_data *event_data)
+{
+	switch (event_data->id) {
+	case LOCATION_EVT_LOCATION:
+		LOG_DBG("Got location:");
+		LOG_DBG("  method: %s", location_method_str(event_data->location.method));
+		LOG_DBG("  latitude: %.06f", event_data->location.latitude);
+		LOG_DBG("  longitude: %.06f", event_data->location.longitude);
+		LOG_DBG("  accuracy: %.01f m", event_data->location.accuracy);
+		LOG_DBG("  altitude: %.01f m", event_data->location.altitude);
+		LOG_DBG("  speed: %.01f m", event_data->location.speed);
+		LOG_DBG("  heading: %.01f m", event_data->location.heading);
+		if (event_data->location.datetime.valid) {
+			LOG_DBG("  date: %04d-%02d-%02d",
+				event_data->location.datetime.year,
+				event_data->location.datetime.month,
+				event_data->location.datetime.day);
+			LOG_DBG("  time: %02d:%02d:%02d.%03d UTC",
+				event_data->location.datetime.hour,
+				event_data->location.datetime.minute,
+				event_data->location.datetime.second,
+				event_data->location.datetime.ms);
+		}
+
+		stats.search_time = (uint32_t)(k_uptime_get() - stats.start_uptime);
```

Comment:
I think this is OK. It keeps it simple, the location module (API) has x amount of seconds to obtain location. Otherwise, a timeout event is expected.

Regarding if we should factor in LTE for GNSS timeouts I _think_ I would prefer respecting the timeout as best as possible. Meaning, not factor in LTE activity. And document that search times for GNSS can be affected by LTE and that the timeout should preferably be increased in poor networks, etc... The location API could even take this into account, if the GNSS search is continuously blocked, it has a greater incentive of using a different method to be able to obtain the location within the time out.

I can imagine users being slightly confused when the location module times out with varying duration. After all, it is a timeout.

Reply:
No plans to change search_time so I'm closing this.

GNSS timeout is discussed elsewhere also.